pkg/dogstatsd/parse.go

```diff
@@ -82,6 +83,41 @@ func (p *parser) parseTags(rawTags []byte) []string {
 	return tagsList
 }
 
+// parse a list of float64 separated by colonSeparator
+func parseFloat64List(rawFloats []byte) ([]float64, error) {
+	var value float64
+	var err error
+	idx := 0
+	values := []float64{}
```

Comment:
Just a note here that this is introducing a new source of allocation (the other one being the tags parsing/handling), which will add some work to the GC and bringing some known caveats.
At some point we may want to use a pool here, it's probably worth benchmarking before adding it though.

Reply:
Good catch, I added a float64_list pool.